arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+# Bucket management in soda-multicloud
+
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10), [Vineela Pachchipulusu]()
+
+
+## Motivation and background
+
+In the cloud world, every cloud vendor provides storage space to users. Generally in terms of bucket and folder. A user can store the data/objects in a specific bucket or folder.
+
+SODA multi-cloud(gelato) is a project which manages multiple cloud vendors from one place. And also interfaces to use cloud services. So it is important for soda to provide bucket management support for various cloud vendors. So that user can create and manage buckets from one place(soda multi-cloud).
+
+## Goals
+The document provides the use case, design and implementation guidelines for supporting bucket management from SODA platform. It includes creation, deletion, object uploading, migration etc. in various cloud providers.
+
+### Non-Goals
+NA
+
+### Requirement Analysis
+
+##### Functional Requirements
+
+1. Create buckets in various cloud providers. Ex: Aws, gcp, azure
```

Comment:
In case of this bucket management, What will be global object managed for Buckets. Will be keep the same model objects as different cloud vendors(Buckets->Folders->Objects) have or the existing model hierarchy in SODA multi-cloud?

Reply:
Now, the hierarchy will be:
1. Bucket->Objects
2. Bucket->Folder->Objects